src/Console/Command/SelfUpdateCommand.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace Pickle\Console\Command;
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Humbug\SelfUpdate\Updater;
+
+class SelfUpdateCommand extends Command
+{
+    const PHAR_NAME = 'pickle.phar';
+
+    protected function configure()
+    {
+        $this
+            ->setName('self-update')
+            ->setDescription('Validate a PECL extension')
+            ->addOption(
+                'unstable',
+                null,
+                InputOption::VALUE_NONE,
+                'Update to a non-stable version'
+            )
+        ;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        if (\Phar::running() === '') {
+            throw new \RuntimeException($this->getName() . ' can only be run from the Phar!');
+        }
+
+        $updater = new Updater(null, false, Updater::STRATEGY_GITHUB);
+        $updater->getStrategy()->setPackageName('friendsofphp/pickle');
+        $updater->getStrategy()->setPharName(self::PHAR_NAME);
+        $updater->getStrategy()->setCurrentLocalVersion('v0.5.0');
+
+        if ($input->getOption('unstable')) {
+            $updater->getStrategy()->setStability('unstable');
+        }
+
+        if ($updater->update() === false) {
```

Comment:
See https://github.com/padraic/humbug/blob/master/src/Command/SelfUpdate.php and README

The update() method returns a boolean. True for updated, and False where no update was available. It throws an Exception on other errors, would typically be signature or HTTP errors. So you should put that line in a `try...catch` block and use the result to tailor the message for either a successful update or no update available, reporting the error in the `catch` as needed.